This fixes several items, see commit comments (#31)
ros-industrial-consortium/tesseract_planning: README.md

### Ubuntu

> ### Recommend (tcmalloc)

Reviewer: ```suggestion
### `tcmalloc` (recommended)
```

> The recommended approach to address this is by leveraging tcmalloc library which is part of google performance tools (gperftools). This has shown to provided more stable memory management when leveraging the planning server.

Reviewer: ```suggestion
The recommended approach to address memory management is to leverage the `tcmalloc` library which is part of Google Performance Tools (`gperftools`). Experimentation has shown `tcmalloc` to provide more stable memory management when leveraging the planning server.
```

> This can be leveraged by linking your exectuable which leverages the planning server against tcmalloc by doing the following:

Reviewer: ```suggestion
This capability can be leveraged by linking the exectuable that contains the planning server against `tcmalloc` using the following commands:
```

Note: This is typically set in the roslaunch file for the planning server node leveraging the `<env name="TCMALLOC_RELEASE_RATE" value="10">` tag.

> ### Default Ubuntu OS malloc (ptmalloc)

Reviewer: ```suggestion
### `ptmalloc` (Ubuntu OS default)
```

> The great thing is, the default malloc library `ptmalloc` used provided run-time [tunables](https://www.gnu.org/software/libc/manual/html_node/Tunables.html), leveraging a set of environment variables.

Reviewer: ```suggestion
`ptmalloc` provides run-time [tunables](https://www.gnu.org/software/libc/manual/html_node/Tunables.html) through a set of environment variables.
```

> The one parameter that has been used to configure libc so it always gives back the memory to the system is setting following memory tunnable:

Reviewer: You should elaborate on why giving memory back to the system is important and why it doesn't do it by default.

Also `tunnable` -> `tunable`

Author: I believe the main paragraph under `Process Planning Server Tuning` explains this, does it not?

See below:

The planning server is built leveraging the library taskflow which enables you to quickly write parallel and heterogeneous tasks programs which aligns well with process motion planning. These types of libraries leverage thread pools to quickly spin up different tasks as needed. One interesting thing found, if a task consumes a lot of memory libc will make a performance decision to allow the thread to hold onto the memory it has for subsequent tasks opposed to giving the memory back to your system. This may be fine if this was the only set of tasks being run but typically in our use case there are other computationally intensive processes which could use this memory.

Reviewer: Okay I think that's fine. I must have skipped over that section

Author: Yea it most likely was not in the changes because I don't think I changed this paragraph.
